tests/Unit/FileTest.php

```diff
@@ -0,0 +1,236 @@
+    {
+        parent::setUp();
+
+        $this->pngFormat = [
+            'mime_type' => 'image/png',
+            'extension' => 'png',
+        ];
+
+        $this->jpegFormat = [
+            'mime_type' => 'image/jpeg',
+            'extension' => 'jpg',
+        ];
+
+        $this->pdfFormat = [
+            'mime_type' => 'application/pdf',
+            'extension' => 'pdf',
+        ];
+
+        $this->stream = $this->createMock(StreamInterface::class);
+        $this->stream
+            ->method('getContents')
+            ->willReturn('This is some test data.');
+
+        $this->base64data = base64_encode('This is some test data.');
+
+        $this->path = tempnam(sys_get_temp_dir(), 'myparcelcom_file') . '.pdf';
+        file_put_contents($this->path, 'This is some test data.');
+    }
+
+    /** @test */
+    public function testId()
+    {
+        $file = new File();
+
+        $this->assertNull($file->getId());
+        $this->assertEquals('4bquri', $file->setId('4bquri')->getId());
+    }
+
+    /** @test */
+    public function testType()
+    {
+        $file = new File();
+
+        $this->assertEquals('files', $file->getType());
+    }
+
+    /** @test */
+    public function testResourceType()
+    {
+        $file = new File();
+
+        $this->assertNull($file->getResourceType());
+        $this->assertEquals('some-resource-type', $file->setResourceType('some-resource-type')->getResourceType());
+    }
+
+    /** @test */
+    public function testFormats()
+    {
+        $file = new File();
+
+        $this->assertEmpty($file->getFormats());
+        $this->assertEquals([$this->jpegFormat], $file->addFormat($this->jpegFormat['mime_type'], $this->jpegFormat['extension'])->getFormats());
+
+        // Setting an array of formats removes all other formats.
+        $this->assertEquals([$this->pdfFormat, $this->pngFormat],
+            $file
+                ->addFormat($this->jpegFormat['mime_type'], $this->jpegFormat['extension'])
+                ->setFormats([$this->pdfFormat, $this->pngFormat])
+                ->getFormats()
+        );
+
+        // Adding a format to an array of existing formats keeps the old ones.
+        // Order of the formats is always pdf->png->jpeg.
+        $this->assertEquals([$this->pdfFormat, $this->pngFormat, $this->jpegFormat],
+            $file
+                ->setFormats([$this->pngFormat, $this->jpegFormat])
+                ->addFormat($this->pdfFormat['mime_type'], $this->pdfFormat['extension'])
+                ->getFormats()
+        );
+    }
+
+    /** @test */
+    public function testStream()
+    {
+        $file = new File();
+
+        $this->assertNull($file->getStream());
+
+        $stream_A = $file
+            ->setStream($this->stream, 'image/png')
+            ->getStream('image/png');
+        $this->assertInstanceOf(StreamInterface::class, $stream_A);
+        $this->assertEquals($this->stream->getContents(), $stream_A->getContents());
```

Comment:
Since you know what you put in the stream, you can check on the value `'This is some test data.'` (or a constant) instead of whatever the stream returns.